### PR TITLE
Chore/hash passwords early

### PR DESCRIPTION
**Pull request by osmaczko**
### What does the PR do

fixes: https://github.com/status-im/status-desktop/issues/11913

### Affected areas

- joining community
- revealing addresses

### CODE REVIEW

File: src/app_service/service/community/service.nim [hunk 1]
```diff
@@ -1478,7 +1479,7 @@ QtObject:
         slot: "onAsyncRequestToJoinCommunityDone",
         communityId: communityId,
         ensName: ensName,
-        password: password,
+        password: if password != "": utils.hashPassword(password) else: "",
```
Comment (jrainville) [inline src/app_service/service/community/service.nim:1482]: Do you think we could hash it even earlier? Like in the module where we get it?

Comment (osmaczko) [reply src/app_service/service/community/service.nim:1482]: Of course we could. I think doing it here is a good trade-off between code duplication and proneness to arguments being logged. We don't do any logging at service level at the moment.

By the way, do you believe that a hashed password is significantly safer than plain text in this exact scenario? Even if a potential attacker reads a hashed password from our logs, they can still use it to log in. That being said, I believe the ultimate solution lies in log levels. Information like function arguments, which might contain passwords (whether in plain-text or hashed), should be at the debug level. Users should be warned about the potential risks before enabling debug-level logging.

Comment (osmaczko) [reply src/app_service/service/community/service.nim:1488]: Issue for the popup: https://github.com/status-im/status-desktop/issues/12308

**Comment by osmaczko** [comment]
@MishkaRogachev @mprakhov friendly review reminder :pray:

**Review by jrainville — APPROVED** [review]

**Review by MishkaRogachev — APPROVED** [review]
Looks good! 
A minor question: there are also some `utils.hashPassword` calls in community tokens and ens. There's no debugging, but maybe we should move them too?

**Comment by osmaczko** [comment]
> Looks good! A minor question: there are also some `utils.hashPassword` calls in community tokens and ens. There's no debugging, but maybe we should move them too?

Good point :+1: Will do

**Comment by osmaczko** [comment]
> > Looks good! A minor question: there are also some `utils.hashPassword` calls in community tokens and ens. There's no debugging, but maybe we should move them too?
> 
> Good point 👍 Will do

@MishkaRogachev done

**Pull request merged**